### PR TITLE
WT-9459 Any table might retain the last pre-crash object on restart.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: test/csuite/tiered_abort/main.c [hunk 1]
```diff
@@ -265,13 +265,13 @@ thread_flush_run(void *arg)
         testutil_check(pthread_rwlock_wrlock(&flush_lock));
         testutil_check(session->flush_tier(session, NULL));
         testutil_check(pthread_rwlock_unlock(&flush_lock));
-        printf("Flush tier %" PRIu32 " completed.\n", i);
+        printf("Flush tier %" PRIu32 " completed.\n", ++i);
```
Comment (ddanderson) [inline test/csuite/tiered_abort/main.c:268]: I'd put the increment in the for loop for this as well.

**Review by ddanderson — APPROVED** [review]
This basically looks good, I had a stylistic concern.

File: test/csuite/tiered_abort/main.c [hunk 2]
```diff
@@ -211,7 +211,7 @@ thread_ckpt_run(void *arg)
      * Keep a separate file with the records we wrote for checking.
      */
     testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
-    for (i = 0;; ++i) {
+    for (i = 0;;) {
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply test/csuite/tiered_abort/main.c:214]: I noticed it was starting with a 0 counter. I made it consistent with schema_abort/main.c, and the others. We could open up a ticket to make all of them the same. But the 0 counter was bugging me.

[the remark this answers is not in the document]
Comment (ddanderson) [reply test/csuite/tiered_abort/main.c:214]: I would change to `for (i = 1;; ++i) {`.  I'd rather not be consistent with a less readable version (assuming the others use `(i = 0;;)` ).  The safety argument is that someone inserts code above the mid-loop increment that uses `i`.  It just makes it harder to look at I think (going through my head: where's the increment?  are there other uses of i?  why was it done this way?).

**Pull request merged**